### ISSUE TITLE
bug: placeOrder backwards incompatible changes included in patch from v2.4.6-p7 -> v2.4.6-p8.

### ISSUE THREAD

**Issue by damienwebdev** [magento/magento2#39282]
As a developer, I want my app to continue working after I **apply security patches**.

### 
1. Be on Magento v2.4.6-p7, with a working application.
2. Upgrade to v2.4.6-p8 on release day because there are security vulnerabilities.
3. [Break existing user-facing workflows](https://github.com/magento/magento2/blame/6d1912708f8888efb6be037386515f566aa0d65a/app/code/Magento/QuoteGraphQl/etc/schema.graphqls#L205) because semver doesn't matter and software is a meme.
4. Wonder how [this commit](https://github.com/magento/magento2/commit/8e3a2e050a6b3a7cf7f3ecefbc14e541b5f23eb9) made it into a security release.

For reproduction:

1. Run `placeOrder` on v2.4.6-p7 and expect an error graphql response.
2. Write code based on the behavior of v2.4.6-p7
3. Upgrade to v2.4.6-p8
4. Run `placeOrder` on v2.4.6-p7 and expect an error graphql response.
5. (BREAKING CHANGE) Get a "success" response with a `null` for data with a new "errors" key.

```json
{
    "data": {
        "placeOrder": {
            "order": null,
            "__typename": "PlaceOrderOutput"
        }
    }
}
```

I'm getting painfully bored and tired of breaking applications because I try to have good security routines like applying security patches the day of release.

https://experienceleague.adobe.com/en/docs/commerce-operations/release/notes/security-patches/2-4-6-patches contains no mention of this BiC change. 

> If a change results in user programs breaking, it's a bug in the kernel. We never EVER blame the user programs. How hard can this be to understand?

Seriously. Do better.

**Comment by JesKingDev** [comment]
+1 This is also happening on `2.4.7-p3`. CC'd from https://github.com/magento/magento2/issues/39300#issuecomment-2448376129

In `2.4.7-p3`, I get no errors at all. Calling `placeOrder` with anything wrong or missing (e.g. billing address, payment, etc.) returns the response
```json
{
    "data": {
        "placeOrder": {
            "order": null
        }
    }
}
```
This is a blocker for any upgrades or security patches. @damienwebdev Thank you for your description that clearly conveys the outrage I feel at the moment, having discovered this issue while finalizing our upgrade to 2.4.7.

**Comment by nwcasebolt** [comment]
Note that this same breaking pattern has already been introduced into the 2.4-develop branch for the UpdateCartItems resolver in [this commit](https://github.com/magento/magento2/commit/9db836a5d29dff01e75189dedc94ec055d88deb0#diff-915fbad5fa41d92498826791e829d3a5ec4384d47432d764d602bdb56471097e). If Adobe does not nip this in the bud now, it's going to spread.

[CC'd from 39300 (comment)](https://github.com/magento/magento2/issues/39300#issuecomment-2452101378)

**Comment by hostep** [comment]
Adobe posted a KB article about this some hours ago and it contains a patch: https://experienceleague.adobe.com/en/docs/commerce-knowledge-base/kb/troubleshooting/known-issues-patches-attached/backwards-incompatible-changes-for-graphql-placeorder-api-in-adobe-commerce-2-4-6-p8

**Comment by damienwebdev** [comment]
Closing as a patch is available.

**Comment by JesKingDev** [comment]
@damienwebdev The patch does not fix the problem in 2.4.7. Should I open a separate issue?

**Comment by damienwebdev** [comment]
@JesKingDev https://github.com/magento/magento2/issues/39300 tracks the issue in 2.4-develop

**Comment by damienwebdev** [comment]
If there's a separate issue introduced as a patch to v2.4.7, then this should be tracked as a separate issue.

However, if this was released in v2.4.7 with no further patches, then I think the other issue suffices.
